### PR TITLE
Implementing Rest Controllers 

### PR DESCRIPTION
**Pull request by eyupbarlas**
| Issue |
|------------|
| Issue #8 |

### :twisted_rightwards_arrows: Branch to merge 
- [x] **master** (i.e. main branch used for deployment and release)

### :loudspeaker: Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (urgent fix for an issue)

### :scroll: Description

> * REST API endpoints are implemented
> * **UNIT TESTS TO ADD!**

### :rotating_light: Unit Tests added or updated as required ?**

- [ ] Not Applicable - Not Needed
- [x] Yes
- [ ] No

### :pencil: Documentation Update required ?

- [ ] Not Applicable - Not Needed
- [X] Yes
- [ ] No

### CODE REVIEW

File: java-backend/src/main/java/com/backend/javabackend/service/UserService.java [hunk 1]
```diff
@@ -21,23 +21,28 @@ public class UserService {
     public List<User> getAllUsers(){
         return userRepository.findAll();
     }
+    public User getUser(Long userId) {
+        return userRepository.findById(userId)
+                .orElseThrow( ()-> new IllegalStateException("User not found with ID: "+userId));
```
Comment (IsmayilMohsumov) [inline java-backend/src/main/java/com/backend/javabackend/service/UserService.java:26]: throw Custom exception

File: java-backend/src/main/java/com/backend/javabackend/rest/UserRest.java [hunk 2]
```diff
@@ -1,18 +1,82 @@
...
 @RestController
 @RequestMapping(value = API_V_1_USER)
 @RequiredArgsConstructor
 public class UserRest {
-    protected final Logger logger = LoggerFactory.getLogger(this.getClass());
+
     private final UserService userService;
+
+    @GetMapping("/users")
+    public ResponseEntity<UserResponse> getUsers() {
```
Comment (IsmayilMohsumov) [inline java-backend/src/main/java/com/backend/javabackend/rest/UserRest.java:30]: https://medium.com/@nadinCodeHat/rest-api-naming-conventions-and-best-practices-1c4e781eb6a5

**Review by IsmayilMohsumov — CHANGES_REQUESTED** [review]

File: java-backend/src/main/java/com/backend/javabackend/service/UserService.java (hunk 1 above)
Comment (eyupbarlas) [reply java-backend/src/main/java/com/backend/javabackend/service/UserService.java:26]: Updated.

File: java-backend/src/main/java/com/backend/javabackend/rest/UserRest.java (hunk 2 above)
Comment (eyupbarlas) [reply java-backend/src/main/java/com/backend/javabackend/rest/UserRest.java:30]: Updated.

**Review by IsmayilMohsumov — DISMISSED** [review]

**Review by IsmayilMohsumov — APPROVED** [review]

**Pull request merged**